pandas/_libs/src/ujson/lib/ultrajsondec.c

```diff
@@ -160,12 +164,35 @@ FASTCALL_ATTR JSOBJ FASTCALL_MSVC decode_numeric(struct DecoderState *ds) {
                 // FIXME: Check for arithmetic overflow here
                 // PERF: Don't do 64-bit arithmetic here unless we know we have
                 // to
-                intValue = intValue * 10ULL + (JSLONG)(chr - 48);
+                newDigit = (JSLONG)(chr - 48);
+
+                // TO DO: need to fix overflow catching
+                if (intValue> (overflowLimit-newDigit)/10) {
```

Comment:
@WillAyd Will this work? (`intValue` is always positive so it's always floor division on the RHS)

Reply:
I don't think so. Assuming a 64 bit platform where the largest integer is 2 ** 63 - 1 this would not branch for 2 ** 63 (which it should)